### PR TITLE
Added Network field to DHT header

### LINKED ISSUE #1231: Add chain_type to DHT message header metadata and validate it against node configuration
Adding the chain type to DHT message header would make compliant wallets/nodes reject messages that erroneously send messages to the wrong chain.

i.e. Wallet A spending a UTXO from mainnet adds Network::Main to a message header.
If wallet B running on testnet somehow receives this transaction, would explicitly reject the message.

On receipt of a message containing chain_type metadata:
The DHT module SHOULD check the chain type flag against it's configuration.
If the node's configuration matches the chain_type, the message should be silently accepted.
If the node's configuration mismatches the chain_type, the node SHOULD reject the message by sending a `DhtRejectMsg` message with the reason `UnsupportedChainType`
The receiver of a DhtRejectMsg message MUST notify the upstream components

### PR DESCRIPTION
**Pull request by sdbondi**
## Description

- Current choice of network: MainNet, TestNet, LocalTest (local only)
- Node will explicitly reject messages from networks other than the one
  for which it is configured.

## Motivation and Context

Closes #1231 

## How Has This Been Tested?

Unit tests for new 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: comms/dht/src/config.rs [hunk 1]
```diff
@@ -89,6 +92,10 @@ impl Default for DhtConfig {
             broadcast_cooldown_max_attempts: 3,
             broadcast_cooldown_period: Duration::from_secs(60 * 30),
             discovery_request_timeout: Duration::from_secs(2 * 60),
+            #[cfg(test)]
```
Comment (CjS77) [inline comms/dht/src/config.rs]: Integration tests will set the `network` field to Testnet.

Maybe just a convenience wrapper around
```
DhtConfig {
  network:  Network::LocalTest,
  .. DhtConfig::default()
}
```
is a better approach

File: comms/dht/src/macros.rs [hunk 2]
```diff
@@ -45,6 +45,7 @@ macro_rules! acquire_write_lock {
 }
 
 #[cfg(any(test, feature = "test-mocks"))]
+#[allow(unused_macros)]
```
Comment (CjS77) [inline comms/dht/src/macros.rs]: You _could_ just delete the unused macro :)

File: comms/dht/src/proto/google.protobuf.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (CjS77) [inline comms/dht/src/proto/google.protobuf.rs]: Is this file required?

File: comms/dht/src/macros.rs (hunk 2 above)
Comment (sdbondi) [reply comms/dht/src/macros.rs]: Haha, it's not really unused and tests will not compile if I remove it. Though the obvious solution is to just not use it in the few places in tests it is used. I'll do that :)

File: comms/dht/src/proto/google.protobuf.rs (hunk 3 above)
Comment (sdbondi) [reply comms/dht/src/proto/google.protobuf.rs]: It's generated by prost because we use google's namespace. Haven't found a way to prevent that - I'll gitignore it.

File: comms/dht/src/config.rs (hunk 1 above)
Comment (sdbondi) [reply comms/dht/src/config.rs]: I've made it use LocalTest for cfg(test) and provided a convenience function for use with integration tests and tests outside of the crate
